src/ood-preview/lib/ood_preview/tutorial.ml

```diff
@@ -0,0 +1,58 @@
+type proficiency =
+  [ `Beginner
+  | `Intermediate
+  | `Advanced
+  ]
+
+type metadata =
```

Comment:
I've never seen this `type t = <some-type-alias> = <type-def>` before, what is the purpose? It has been used for the other entries (e.g. `Success_story`), should this be 

```ocaml
type metadata = Tutorial.t = ...
```
Could we not just pull in the information from the library in `ood-cli` which has the types or is it so you can rename the type to `metadata` and produce derivers for this new type name?

Reply:
In this case, I use it to generate the YAML converters functions with the ppx.

The problem is, the yaml ppx deriver does not seem to support the `[@key ""]` (or is it `[@name ""]` attribute for polymorphic variants, so I need to use a string instead and convert to the polymorphic variant manually. That's why I didn't add the `=  Tutorial.t` in this case, because the types are not actually equal.

To share a few thoughts related to this question: I was wondering why the types in `Ood` don't include the `body` field?
Also, we could make `Ood` depend on `ppx_deriving_yaml`, I think the ppx syntax will just be ignored by ReScript, right? That would make the types in `Ood` much more reusable in `Ood_preview`.

Re: "Could we not just pull in the information from the library in ood-cli which has the types"
I didn't look much into `ood-cli`. I'm not sure which types we would pull in, because we use already defined types in `Ood` (with the exception of `Tutorial.metadata` here). What would be the benefit of depending on `ood-cli`? One immediate drawback is that `ood-cli` depends on a lot of irrelevant things for the `ood-preview`, e.g. the netlify stuffs.

Reply:
Yeah the variant thing is annoying, I have an issue related to it https://github.com/patricoferris/ppx_deriving_yaml/issues/23

I don't think there is any particular reason they don't include the body, it was just for clearer distinction between metadata and markdown body. Maybe it is possible to share some of this across the collections similar to [what I did in sesame](https://github.com/patricoferris/sesame/blob/main/lib/collection.mli#L12-L27) as only the metadata changes between them.

> Also, we could make Ood depend on ppx_deriving_yaml, I think the ppx syntax will just be ignored by ReScript, right? That would make the types in Ood much more reusable in Ood_preview.

I don't know enough about ReScript but I think the problem is `ppx_deriving_yaml` needs `Yaml` which needs C-bindings and at that point things break. If we could somehow fix this and have single unified set of types I would be happy.

> I didn't look much into ood-cli. I'm not sure which types we would pull in, because we use already defined types in Ood (with the exception of Tutorial.metadata here)

What about https://github.com/ocaml/ood/blob/main/src/ood-cli/data.ml ? I just realised the last PR completely removed `lib_ood` which is what I was really referring to here. But I think you are right about dependencies.

Reply:
> I don't know enough about ReScript but I think the problem is ppx_deriving_yaml needs Yaml which needs C-bindings and at that point things break.

I was thinking that the deriver would simply be ignored and no code depending on Yaml would be generated for ReScript. In OCaml, this would generate that the other libs depend on.

But now that I'm writing this, it seems like a very obscure and confusing setup, so probably not a good idea.

> I don't think there is any particular reason they don't include the body, it was just for clearer distinction between metadata and markdown body.

Could we add types for `metadata` and `t` in this case? where `t` would be 

```ocaml
type t = {
  meta: metadata;
  body: string
}
```

>  just realised the last PR completely removed lib_ood which is what I was really referring to here.

Oops, I should have mentioned it in the PR more explicitly. I did this to simplify the setup a bit, but if you believe there was a use for it being provided as a lib, I'll revert this :)

Reply:
If you're ok, I'll merge the PR without reworking the types. It seems like an important conversation, so I'd prefer to have it in its own PR, that I can open once this is merged.